### PR TITLE
adds missing tests to Page and its sub-components

### PR DESCRIPTION
**Pull request by kvendrik**
### WHY are these changes introduced?
fixes https://github.com/Shopify/polaris-react-deprecated/issues/2438

### WHAT is this pull request doing?
- Adds missing tests to Page and its sub-components.
- Removes indicator logic (more details here https://github.com/Shopify/polaris-react/pull/526/commits/ca283f3b8dad9090b2308831a3a89542345c3373).

### CODE REVIEW

**Comment by danrosenthal** [comment]
@kvendrik is this still a WIP, or is it ready to be reviewed? Looks like the `WIP` bot is picking up a work in progress from a linked PR.

**Comment by kvendrik** [comment]
@danrosenthal this is ready to be review, tagged @solonaarmstrong last week but looks like she hasn't had time yet. 😄

**Comment by solonaarmstrong** [comment]
> You make writing tests look easy

That he does!

File: src/components/Page/components/Header/components/Action/tests/Action.test.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import {Icon, UnstyledLink} from 'components';
```
Comment (solonaarmstrong) [inline src/components/Page/components/Header/components/Action/tests/Action.test.tsx]: I'm not sure we can import this way?

File: src/components/Page/components/Header/components/Action/tests/Action.test.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  describe('disabled', () => {
+    it('gets passed into the button', () => {
+      const action = mountWithAppProvider(<Action disabled />);
+      expect(action.find('button').prop('disabled')).toBeTruthy();
+    });
+  });
+
+  describe('children', () => {
+    it('get rendered when an icon is present', () => {
```
Comment (solonaarmstrong) [inline src/components/Page/components/Header/components/Action/tests/Action.test.tsx]: gets?

File: src/components/Page/components/Header/components/Action/tests/Action.test.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+  describe('onAction()', () => {
+    it('triggers when the button gets clicked', () => {
+      const onActionSpy = jest.fn();
+      const action = mountWithAppProvider(<Action onAction={onActionSpy} />);
+      trigger(action.find('button'), 'onClick');
+      expect(onActionSpy).toHaveBeenCalledTimes(1);
+    });
+  });
+});
```
Comment (solonaarmstrong) [inline src/components/Page/components/Header/components/Action/tests/Action.test.tsx]: Should this have a test for `disclosure`?

File: src/components/Page/components/Header/components/ActionGroup/ActionGroup.tsx [hunk 4]
```diff
@@ -20,11 +17,9 @@ class ActionGroup extends React.Component<Props, never> {
   render() {
     const {actions, details, title, icon, active} = this.props;
 
-    const detailsMarkup = details ? (
+    const detailsMarkup = details && (
```
Comment (solonaarmstrong) [inline src/components/Page/components/Header/components/ActionGroup/ActionGroup.tsx]: Why this change?

File: src/components/Page/components/Header/components/Action/tests/Action.test.tsx (hunk 3 above)
Comment (kvendrik) [reply src/components/Page/components/Header/components/Action/tests/Action.test.tsx]: There is a test that makes sense the `children` get rendered when disclosure if defined (because of the if statement that controls how they get rendered based on if `disclosure` is defined). Apart from that the only related behaviour is the addition of an icon which is purely presentational and we therefor want to leave to the VRT.

File: src/components/Page/components/Header/components/ActionGroup/ActionGroup.tsx (hunk 4 above)
Comment (kvendrik) [reply src/components/Page/components/Header/components/ActionGroup/ActionGroup.tsx]: Wasn't necessary just came past is and decided to clean it up. 😄

Comment (solonaarmstrong) [reply src/components/Page/components/Header/components/ActionGroup/ActionGroup.tsx]: I'm not clear on when we use `&&`. Wouldn't this need to be null if it doesn't pass the condition, since it's markup?

Comment (kvendrik) [reply src/components/Page/components/Header/components/ActionGroup/ActionGroup.tsx]: [Its explained more in-depth in React's docs](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator) but the short answer is that `undefined` is part of type `ReactNode`. 😄
